x/tally/keeper/filter.go

```diff
@@ -39,34 +50,46 @@ func ApplyFilter(input []byte, reveals []types.RevealBody) ([]int, bool, []strin
 		}
 	}
 	if maxFreq*3 < len(reveals)*2 {
-		return make([]int, len(reveals)), false, nil, types.ErrNoBasicConsensus
+		return result, types.ErrNoBasicConsensus
+	}
+	result.ProxyPubKeys = proxyPubKeys
+
+	params, err := k.GetParams(ctx)
+	if err != nil {
+		return result, err
 	}
 
 	var filter types.Filter
-	var err error
 	switch input[0] {
 	case filterTypeNone:
 		filter, err = types.NewFilterNone(input)
+		result.GasUsed = params.FilterGasCostNone
 	case filterTypeMode:
 		filter, err = types.NewFilterMode(input)
+		result.GasUsed = params.FilterGasCostMultiplierMode * uint64(replicationFactor)
```

Comment:
Is there a reason the replication factor is an `int64`? On the Rust side it's a `u16`.

Reply:
ah didn't realize it. updated to `uint16`